lib/eslint.js

```diff
@@ -617,33 +657,45 @@ module.exports = (function() {
             Object.keys(config.rules).filter(function(key) {
                 return getRuleSeverity(config.rules[key]) > 0;
             }).forEach(function(key) {
-
-                var ruleCreator = rules.get(key),
-                    severity = getRuleSeverity(config.rules[key]),
-                    options = getRuleOptions(config.rules[key]),
-                    rule;
-
-                if (ruleCreator) {
-                    try {
-                        rule = ruleCreator(new RuleContext(
-                            key, api, severity, options,
-                            config.settings, config.ecmaFeatures
-                        ));
-
-                        // add all the node types as listeners
-                        Object.keys(rule).forEach(function(nodeType) {
-                            api.on(nodeType, timing.enabled
-                                ? timing.time(key, rule[nodeType])
-                                : rule[nodeType]
-                            );
-                        });
-                    } catch(ex) {
-                        ex.message = "Error while loading rule '" + key + "': " + ex.message;
-                        throw ex;
+                var ruleCreator,
+                    severity,
+                    options,
+                    rule,
+                    deprecationMsg;
+
+                if (!rules.get(key)) {
```

Comment:
I guess I'm confused about what we're trying to do here then.  I thought the idea for `v1.0.0` was that the rules and their tests would be entirely removed.  If we're talking about a scenario where the rules are still in place but are due to be removed at a future date, then maybe the best thing to do is what you suggest, for the rule to be modified and something like the following added to the rest of the rule:

```js
Program: function(node) {
    context.report(node, "This rule will be deprecated in vX.Y.Z, it will be replaced by: 'other-rule');
}
```

Would that work?